snaps_boot/provision/hardware/pxe_utils.py

```diff
@@ -557,9 +575,10 @@ def __provision_clean(proxy_dict):
     os.system('apt-get  -y remove isc-dhcp-server')
     logger.info("unmount mount point")
     os.system('umount  /mnt')
-    
-    if (proxy_dict["ngcacher_proxy"] <> ""):
-      __clean_ngcacher_proxy(proxy_dict)
+
+    if proxy_dict.get('ngcacher_proxy'):
```

Comment:
This line is where the actual bug fix is

Reply:
My python is fuzzy on this, but  if ngcacher_proxy = "" does this run __clean_ngcacher or not.  
the way I wrote it in issue-90 was ```if proxy_dict.get('ngcacher_proxy') is None:  proxy_dict['ngcacher_proxy'] = ""``` so if you leave it out it adds it as empty and continues on.  Does this have the same effect?

Reply:
nice catch @dschrimpsher - I'll add a check for an empty string.